pkg/otlp/metrics/kafka_mapping.go

```diff
@@ -435,3 +427,11 @@ func remapJvmMetrics(all pmetric.MetricSlice, m pmetric.Metric) {
 		)
 	}
 }
+
+// renameKafkaMetrics renames otel kafka metrics to avoid conflicts with DD metrics.
+func renameKafkaMetrics(m pmetric.Metric) {
+	switch m.Name() {
+	case "kafka.producer.request-rate", "kafka.producer.response-rate", "kafka.producer.request-latency-avg", "kafka.consumer.fetch-size-avg", "kafka.producer.compression-rate", "kafka.producer.record-retry-rate", "kafka.producer.record-send-rate", "kafka.producer.record-error-rate":
```

Comment:
nit: Here you can store all these constant names in a map then do a map lookup. It probably has better performance than switch-case as the list of names grows.

https://stackoverflow.com/questions/29566229/go-switch-string-efficiency